util/codec/bytes.go

```diff
@@ -122,6 +122,27 @@ func DecodeBytesDesc(b []byte) ([]byte, []byte, error) {
 	return decodeBytes(b, true)
 }
 
+// EncodeCompactBytes joins bytes with its length into a byte slice. It is more
+// efficient in both space and time compare to EncodeBytes.
+func EncodeCompactBytes(b []byte, data []byte) []byte {
+	b = EncodeVarint(b, int64(len(data)))
```

Comment:
oh, I forget it, I thought MaxVarintLen is 9 before. extra 10 or 3 bytes has no big different.